plotting/sound.py

```diff
@@ -65,16 +65,91 @@ def plot(self):
         # Plot Sound Speed profile
         plt.subplot(gs[:, 1 if self.showmap else 0])
         ax = plt.gca()
+        i = 0
         for i, ss in enumerate(self.sspeed):
             ax.plot(ss, self.temperature_depths[i], '-')
 
         minspeed = np.amin(self.sspeed)
         maxspeed = np.amax(self.sspeed)
 
+
+        if self.query.get('annotate'):
+            # SOUND SPEED MINIMA
+            
+            minpos = np.where(self.sspeed[0] == minspeed)
+
+            # SONIC LAYER DEPTH
+
+            # Calculate
+            soniclayerdepth_value = self.sspeed[0][0:int(minpos[0])].max()
+            soniclayerdepth_idx = np.where(self.sspeed[0] == soniclayerdepth_value)
+            soniclayerdepth = self.temperature_depths.data[0][soniclayerdepth_idx][0]
```

Comment:
Could this be wrapped into multiple functions? Maybe something like `self.__calc_soniclayerdepth_value` and `self.__calc_soniclayerdepth`?

Reply:
I think I took a different approach in later calculations.
I created a function which finds the index, and that allows you to find both the soniclayerdepth, and value in a single line

Reply:
So I changed all 4 derived products to use the find_###_idx() functions I created.
Does this satisfy your concerns? Often times the value is not needed.